src/confighttp.cpp

```diff
@@ -316,13 +315,16 @@ void saveApp(resp_https_t response, req_https_t request) {
     pt::read_json(ss, inputTree);
     pt::read_json(config::stream.file_apps, fileTree);
 
-    // Moonlight checks the id of an item to determine if an item was changed
-    // Needs to be a 32-bit positive integer due to client limitations, "0" indicates no app
-    auto id = util::generate_int32(1, std::numeric_limits<std::int32_t>::max());
-    while(ids.count(std::to_string(id)) > 0) {
-      id = util::generate_int32(1, std::numeric_limits<std::int32_t>::max());
+    // Moonlight checks the id of an item to determine if an item was changed, recompute
+    auto possible_ids = proc::calculate_app_id(inputTree.get<std::string>("name"), inputTree.get<std::string>("image-path"), inputTree.get<int>("index"));
+    if(ids.count(std::get<0>(possible_ids)) == 0) {
+      // Avoid using index to generate id if possible
+      inputTree.put("id", std::get<0>(possible_ids));
+    }
+    else {
+      // Fallback to include index on collision
+      inputTree.put("id", std::get<1>(possible_ids));
```

Comment:
This ID written into apps.json isn't used anymore, is it? The code in `parse()` ignores it and uses `calculate_app_id()` instead (as it should). We should remove this and just not write an ID field into apps.json at all.

Reply:
Yeah, I forgot we just follow the same `parse()` logic after this, I'll remove it